### DISCUSSION TITLE
5.x to 6.x upgrade

### DISCUSSION THREAD

**Discussion by gjblajian** [dolanmiu/docx#899] [Q&A] [answered]
Do you have a list of changes or migration suggestions for upgrading from version 5.x to 6.x?

**Comment by dolanmiu** [comment, marked as answer]
Hi

Yup, all of the changes are under the release notes:

https://github.com/dolanmiu/docx/releases

**Reply by dolanmiu** [reply]
The biggest change is now the `sections` is property is now declarative

There is no such `addSection` method anymore

**Reply by gjblajian** [reply]
@dolanmiu I have code that adds sections small pieces at a time based on input I am receiving. How do I add content at the end of the sections array incrementally? All of your examples seem to place all the content in the document in one call.

**Reply by dolanmiu** [reply]
The trick is to prepare the data as an array

You need to think declaratively. Detach yourself from the confines of manipulating arrays, and think one level above

Don't tell it HOW you want it (array push, shift, pop), tell it WHAT you want (data transform to section)

```ts
sections: yourData.map((d) => ({
    children: [
        new Paragraph(d),
    ],
}),
```

So with this, you could gather up all your data, then in the final step, do a `.map` on the data to transform it to section
